### ISSUE TITLE
Option to run multiple fits for infer_rnn_rates and select best one

### ISSUE THREAD

**Issue by abuzarmahmood** [katzlabbrandeis/blech_clust#327]
Probably a good default value to start would be 4

**Comment by abuzarmahmood** [comment]
The issue involves adding an option to run multiple fits for the `infer_rnn_rates` function and selecting the best one. The following suggestions and changes were made:

1. **Relevant File Identified:**
   - **File**: `/home/abuzarmahmood/projects/blech_clust/utils/infer_rnn_rates.py`
   - **Description**: This module uses an Auto-regressive Recurrent Neural Network (RNN) to infer firing rates from electrophysiological data. It processes data for each taste separately, trains an RNN model, and saves the predicted firing rates and latent factors.

2. **Suggested Changes:**
   - **Add a Command-Line Argument for Number of Fits:**
     Modify the `argparse` section to include an argument for specifying the number of fits to run.
     ```python
     parser.add_argument('--num_fits', type=int, default=4,
                         help='Number of fits to run and select the best one (default: %(default)s)')
     ```

   - **Modify the Training Loop to Run Multiple Fits:**
     Update the training loop to run multiple fits and select the best one based on a defined criterion, such as the lowest loss.
     ```python
     best_loss = float('inf')
     best_model = None

     for fit in range(args.num_fits):
         print(f'Running fit {fit + 1}/{args.num_fits}')
         model, loss = train_model(spike_data, params_dict)  # Assuming train_model returns model and loss

         if loss < best_loss:
             best_loss = loss
             best_model = model
             print(f'New best model found with loss: {best_loss}')

     # Save the best model
     torch.save(best_model.state_dict(), model_save_path)
     ```

   - **Ensure the `train_model` Function Supports Multiple Fits:**
     Ensure that the `train_model` function can return the loss or any other metric used to evaluate the model's performance.

   - **Update the Model Saving Logic:**
     Ensure that only the best model is saved after all fits are completed.

These changes will allow the script to run multiple fits and select the best one based on the specified criterion. Testing these changes thoroughly is recommended to ensure they work as expected.

---
*This response was automatically generated by blech_bot*
